tests/Core/Util/ExitCode/ExitCodeTest.php

```diff
@@ -0,0 +1,677 @@
+     * @param int           $expected  The expected exit code.
+     *
+     * @dataProvider dataPhpcs
+     *
+     * @return void
+     */
+    public function testPhpcsNoParallel($extraArgs, $expected)
+    {
+        $extraArgs[] = self::SOURCE_DIR.'mix-errors-warnings.inc';
+
+        $this->runPhpcsAndCheckExitCode($extraArgs, $expected);
+
+    }//end testPhpcsNoParallel()
+
+
+    /**
+     * Verify generated exit codes (PHPCS) when using parallel processing.
+     *
+     * Note: there tests are skipped on Windows as the PCNTL extension needed for parallel processing
+     * isn't available on Windows.
+     *
+     * @param array<string> $extraArgs Any extra arguments to pass on the command line.
+     * @param int           $expected  The expected exit code.
+     *
+     * @dataProvider dataPhpcs
+     * @requires     OS ^(?!WIN).*
+     *
+     * @return void
+     */
+    public function testPhpcsParallel($extraArgs, $expected)
+    {
+        // Deliberately using `parallel=3` to scan 5 files to make sure that the results
+        // from multiple batches get recorded and added correctly.
+        $extraArgs[] = self::SOURCE_DIR;
+        $extraArgs[] = '--parallel=3';
+
+        $this->runPhpcsAndCheckExitCode($extraArgs, $expected);
+
+    }//end testPhpcsParallel()
+
+
+    /**
+     * Verify generated exit codes (PHPCS) when caching of results is used.
+     *
+     * @param array<string> $extraArgs Any extra arguments to pass on the command line.
+     * @param int           $expected  The expected exit code.
+     *
+     * @dataProvider dataPhpcs
+     *
+     * @return void
+     */
+    public function testPhpcsWithCache($extraArgs, $expected)
+    {
+        $extraArgs[] = self::SOURCE_DIR.'mix-errors-warnings.inc';
+        $extraArgs[] = '--cache='.self::CACHE_FILE;
+
+        // Make sure we start without a cache.
+        if (method_exists($this, 'assertFileDoesNotExist') === true) {
+            $this->assertFileDoesNotExist(self::CACHE_FILE);
+        } else {
+            // PHPUnit < 9.1.0.
+            $this->assertFileNotExists(self::CACHE_FILE);
+        }
+
+        // First run with these arguments to create the cache.
+        $this->runPhpcsAndCheckExitCode($extraArgs, $expected);
+
+        // Check that the cache file was created.
+        $this->assertFileExists(self::CACHE_FILE);
+
+        // Second run to verify the exit code is the same when the results are taking from the cache.
+        $this->runPhpcsAndCheckExitCode($extraArgs, $expected);
+
+    }//end testPhpcsWithCache()
+
+
+    /**
+     * Test Helper: run PHPCS and verify the generated exit code.
+     *
+     * @param array<string> $extraArgs Any extra arguments to pass on the command line.
+     * @param int           $expected  The expected exit code.
+     *
+     * @dataProvider dataPhpcs
+     *
+     * @return void
+     */
+    public function runPhpcsAndCheckExitCode($extraArgs, $expected)
```

Comment:
Is there a reason for this help method and `runPhpcbfAndCheckExitCode()` to be public?

Reply:
No ;-) Copy/pasta from when I split the method off from the original test. I'll make these `private` now.